app/code/Magento/Backend/Model/Search/Config/Result/Builder.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Model\Search\Config\Result;
+
+use Magento\Config\Model\Config\Structure\Element\AbstractComposite;
+use Magento\Config\Model\Config\Structure\ElementInterface;
+
+/**
+ * Config SearchResult Builder
+ */
+class Builder
+{
+    const STRUCTURE_ELEMENT_TYPE_SECTION = 'section';
+    const STRUCTURE_ELEMENT_TYPE_GROUP   = 'group';
+    const STRUCTURE_ELEMENT_TYPE_FIELD   = 'field';
+
+    /**
+     * @var array
+     */
+    private $supportedElementTypes = [
+        self::STRUCTURE_ELEMENT_TYPE_SECTION,
+        self::STRUCTURE_ELEMENT_TYPE_GROUP,
+        self::STRUCTURE_ELEMENT_TYPE_FIELD,
+    ];
```

Comment:
it's better not to hard code supported elements, but accept an additional parameter, which is configured through the DI.

So, if some new supported component would be added in future we could easily add this new type with DI configuration.